hive-metastore/src/main/java/org/apache/iceberg/hive/ClientPool.java

```diff
@@ -54,7 +54,7 @@
       return action.run(client);
 
     } catch (Exception exc) {
-      if (reconnectExc.isInstance(exc)) {
+      if (reconnectExc.isInstance(exc) || failureDetection(exc)) {
```

Comment:
I think it would be more easy to understand if we move `reconnectExc.isInstance(exc)` into the `failureDetection(exc)` method as well.

What do you think?

Reply:
Yes, it can be done, but the constructor of `ClientPool` may change. Because the members of `reconnectExc` will be deleted.   Does this affect other component calls？

Reply:
Or just move reconnectExc.isInstance(exc) into the failureDetection(exc) method in `ClientPool`, like this:
```
try {
      return action.run(client);

    } catch (Exception exc) {
      if (failureDetection(exc)) {
        ...
      }
    }

  protected boolean failureDetection(Exception exc) {
    return reconnectExc.isInstance(exc);
  }
```
@pvary

Reply:
@Jecarm: This is what I have been thinking about too.

And in `HiveClientPool` we can do something like this:
```
  @Override
  protected boolean failureDetection(Exception e) {
    if (super.failureDetection(e) || (e != null && e instanceof MetaException &&
            e.getMessage().contains("Got exception: org.apache.thrift.transport.TTransportException"))) {
      return true;
    }
    return false;
  }
```

Reply:
Okay.